extensions/interactions/NumericInput/NumericInputValidationService.js

```diff
@@ -17,18 +17,101 @@
     return {
       getCustomizationArgsWarnings: function(customizationArgs) {
-        // TODO(juansaba): Implement customization args validations.
         return [];
       },
       getAllWarnings: function(
           stateName, customizationArgs, answerGroups, defaultOutcome) {
-        return this.getCustomizationArgsWarnings(customizationArgs).concat(
+        var warningsList = [];
+
+        warningsList = warningsList.concat(
+          this.getCustomizationArgsWarnings(customizationArgs));
+
+        /*
+        Store an answer range for every rule, then check for redundant
+        ranges. A range is an object of the form:
+        {
+          lb: float, lower bound
+          ub: float, upper bound
+          lbi: bool, is lower bound inclusive
+          ubi: bool, is upper bound inclusive
+        }
+        */
+        var isEnclosedBy = function(ra, rb) {
+          // Checks if range ra is enclosed by range rb.
+          var lowerBoundConditionIsSatisfied =
+              (rb.lb < ra.lb) || (rb.lb == ra.lb && (!ra.lbi || rb.lbi));
+          var upperBoundConditionIsSatisfied =
+              (rb.ub > ra.ub) || (rb.ub == ra.ub && (!ra.ubi || rb.ubi));
+          return lowerBoundConditionIsSatisfied &&
+              upperBoundConditionIsSatisfied;
+        };
+
+        var ranges = [];
+        for (var i = 0; i < answerGroups.length; i++) {
+          var rules = answerGroups[i].rules;
+          for (var j = 0; j < rules.length; j++) {
+            var rule = rules[j];
+            var range = {};
+            switch (rule.type) {
+              case 'Equals':
+                var x = rule.inputs.x;
+                range = { lb: x, ub: x, lbi: true, ubi: true };
+                break;
+              case 'IsInclusivelyBetween':
+                var a = rule.inputs.a;
+                var b = rule.inputs.b;
+                range = { lb: a, ub: b, lbi: true, ubi: true };
+                break;
+              case 'IsGreaterThan':
+                var x = rule.inputs.x;
+                range = { lb: x, ub: Infinity, lbi: false, ubi: false };
+                break;
+              case 'IsGreaterThanOrEqualTo':
+                var x = rule.inputs.x;
+                range = { lb: x, ub: Infinity, lbi: true, ubi: false };
+                break;
+              case 'IsLessThan':
+                var x = rule.inputs.x;
+                range = { lb: -Infinity, ub: x, lbi: false, ubi: false };
+                break;
+              case 'IsLessThanOrEqualTo':
+                var x = rule.inputs.x;
+                range = { lb: -Infinity, ub: x, lbi: false, ubi: true };
+                break;
+              case 'IsWithinTolerance':
+                var x = rule.inputs.x;
+                var tol = rule.inputs.tol;
+                range = { lb: x - tol, ub: x + tol, lbi: true, ubi: true };
+                break;
+              default:
+            }
+            range.answerGroup = i + 1;
```

Comment:
Can we call these `answerGroupIndex` and `ruleIndex` respectively?

On line 58, let's also say:
```
    var range = {
      answerGroupIndex: i + 1,
      ruleIndex: j = 1,
      lb: null,
      ub: null,
      ...
    };
```
and define a function setLowerAndUpperBounds(range, lb, ub, lbi, ubi) to update the given range object. That way, the keys in the object are stable over time.